lib/core/src/Cardano/Wallet/Network.hs

```diff
@@ -287,3 +304,34 @@ follow nl tr cps yield rollback header =
                         "Failed to roll backward: " <> show e
                 Right () -> do
                     step cursor'
+
+onSameTip :: BlockHeader -> BlockHeader -> Bool
+onSameTip = (==)
+
+data ErrInconsistentTips
+    = ErrInconsistentTips
```

Comment:
The type name should rather be `ErrAtomically` here. Although, we should be careful about calling this `atomically`.  Things are not atomic here, but, so long as we do read-only calls, there shouldn't be any issue. Any mutations happening in the `ExceptT` will actually happen, and might happen more than once.